### PR TITLE
Add reduced range and id construction helper 

### PR DESCRIPTION
**Pull request by yuriykoch**
It's easier to store range or id for tests for multiple dimensions as a single 3D range or id respectively

This PR depends on:
#99 - 3fb3d4e - Uses id construction helper

Please, review changes starting from bf5a0f6 - "Add reduced range construction helper"

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Comment by yuriykoch** [comment]
All dependencies are merged. PR contains only relevant changes now.

File: tests/common/get_cts_object.h [hunk 1]
```diff
@@ -183,6 +183,15 @@ struct get_cts_object::range<1> {
   static cl::sycl::range<1> get(size_t r0, size_t, size_t) {
     return cl::sycl::range<1>(r0);
   }
+
+  /**
+   * @brief Constructs a range<1> by using any bigger range
```
[the remark this answers is not in the document]
Comment (yuriykoch) [reply tests/common/get_cts_object.h]: Thanks, that's much clearer. Applied as cfd59ec along with the similar suggestion

File: tests/common/get_cts_object.h [hunk 2]
```diff
@@ -208,6 +217,15 @@ struct get_cts_object::range<2> {
   static cl::sycl::range<2> get(size_t r0, size_t r1, size_t) {
     return cl::sycl::range<2>(r0, r1);
   }
+
+  /**
+   * @brief Constructs a range<2> by using any bigger range
```
[the remark this answers is not in the document]
Comment (yuriykoch) [reply tests/common/get_cts_object.h]: Applied as cfd59ec

File: tests/common/get_cts_object.h [hunk 3]
```diff
@@ -218,7 +236,7 @@ struct get_cts_object::range<2> {
   template <size_t totalSize>
   static cl::sycl::range<2> get_fixed_size(size_t r0, size_t) {
     assert("Parameters passed for fixed size range are not supported" &&
-           (totalSize % r0 == 0))
+           (totalSize % r0 == 0));
```
[the remark this answers is not in the document]
Comment (bader) [reply tests/common/get_cts_object.h:241]: No instantiations of this template?

**Review by keryell — APPROVED** [review]

**Pull request merged**